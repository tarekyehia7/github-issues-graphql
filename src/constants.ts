export const constants = {
	repository: 'facebook',
	projectName: 'react',
	searchTerm: 'title,body',
	issuesPerPage: 11,
	githubUrl: 'https://github.com',
};
