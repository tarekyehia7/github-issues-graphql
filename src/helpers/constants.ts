export const constants = {
    repository: 'facebook',
    projectName: 'react',
	searchTerm: 'title,body',
    issuesPerPage: 10,
    githubUrl: 'https://github.com'
};