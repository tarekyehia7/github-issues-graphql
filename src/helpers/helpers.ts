import formatDistanceToNowStrict from 'date-fns/formatDistanceToNowStrict';

export const getPagesNumber = (totalIssuesCount: number, issuesPerPage: number): number => {
	const totalPages = Math.ceil(totalIssuesCount / issuesPerPage);
	return !totalPages ? 1 : totalPages;
};

export const formatDate = (date: string) => {
	try {
		const myDate = new Date(date);
		const myFormattedDate = formatDistanceToNowStrict(myDate, { addSuffix: true });
		return myFormattedDate;
	} catch (e) {
		console.error(e);
		return '';
	}
};
