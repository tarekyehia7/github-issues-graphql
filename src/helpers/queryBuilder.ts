import { StatusEnum } from "../pages/Issues/Issues";

interface Query {
    repo: string;
    input: string;
    is: string;
    in: string;
    status: StatusEnum;
}

export const buildQuery = (query: Query) => {
    return Object.entries(query)
        .reduce((str, [p, val]) => `${str}${p}:${val} `, '')
        .replace('status:', '')
        .replace('input:', '')
        .trim();
};