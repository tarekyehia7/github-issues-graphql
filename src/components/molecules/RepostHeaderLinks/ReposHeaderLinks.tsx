import React from 'react';

import { constants } from '../../../constants';
import { Link, LinkType } from '../../atoms/link/Link';
import { HeaderLinks, BookIconStyled, RepoStateLabel } from './ReposHeaderLinks.styled';

export const ReposHeaderLinks = () => {
	const { githubUrl, repository, projectName } = constants;

	return (
		<HeaderLinks>
			<BookIconStyled />
			<strong>
				&nbsp;
				<Link type={LinkType.NormalBlue} to={`${githubUrl}/${repository}`}>
					{' '}
					{repository}{' '}
				</Link>
				&nbsp;
			</strong>
			/
			<strong>
				&nbsp;
				<Link type={LinkType.NormalBlue} to={`${githubUrl}/${repository}/${projectName}`}>
					{projectName}
				</Link>
				&nbsp;
			</strong>
			<RepoStateLabel>Public</RepoStateLabel>
		</HeaderLinks>
	);
};
