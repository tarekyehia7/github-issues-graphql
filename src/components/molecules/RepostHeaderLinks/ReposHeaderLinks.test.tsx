import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';

import { ReposHeaderLinks } from './ReposHeaderLinks';
import { PageWithTheme } from '../../../helpers/testing/helpers';
import { constants } from '../../../constants';

const ReposHeaderWithTheme = () => {
	return (
		<PageWithTheme>
			<ReposHeaderLinks />
		</PageWithTheme>
	);
};

describe('<RposHeaderLinks />', () => {
	it('Should match snapshot', () => {
		const { container } = render(<ReposHeaderWithTheme />);
		expect(container).toMatchSnapshot();
	});

	it('Should have correct links', () => {
		const { githubUrl, repository, projectName } = constants;
		const { getAllByRole } = render(<ReposHeaderWithTheme />);
		const links = getAllByRole('link');

		expect(links).toHaveLength(2);
		expect(links[0]).toHaveAttribute('href', `${githubUrl}/${repository}`);
		expect(links[1]).toHaveAttribute('href', `${githubUrl}/${repository}/${projectName}`);
	});
});
