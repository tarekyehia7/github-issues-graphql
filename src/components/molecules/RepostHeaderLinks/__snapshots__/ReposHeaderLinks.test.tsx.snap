// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RposHeaderLinks /> Should match snapshot 1`] = `
.c2 {
  color: #0969da;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #0969da;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-bottom: 1rem;
}

.c1 {
  height: 1rem;
  width: 1rem;
  margin-top: 0.1rem;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 0 7px;
  font-size: 12px;
  font-weight: 500;
  line-height: 18px;
  white-space: nowrap;
  border: 1px solid transparent;
  border-radius: 2em;
  color: #57606a;
  border-color: #d0d7de;
}

<div>
  <div
    class="c0"
  >
    <svg
      aria-hidden="true"
      class="c1"
      data-view-component="true"
      version="1.1"
      viewBox="0 0 16 16"
    >
      <path
        d="M2 2.5A2.5 2.5 0 014.5 0h8.75a.75.75 0 01.75.75v12.5a.75.75 0 01-.75.75h-2.5a.75.75 0 110-1.5h1.75v-2h-8a1 1 0 00-.714 1.7.75.75 0 01-1.072 1.05A2.495 2.495 0 012 11.5v-9zm10.5-1V9h-8c-.356 0-.694.074-1 .208V2.5a1 1 0 011-1h8zM5 12.25v3.25a.25.25 0 00.4.2l1.45-1.087a.25.25 0 01.3 0L8.6 15.7a.25.25 0 00.4-.2v-3.25a.25.25 0 00-.25-.25h-3.5a.25.25 0 00-.25.25z"
      />
    </svg>
    <strong>
       
      <a
        class="c2"
        href="https://github.com/facebook"
      >
         
        facebook
         
      </a>
       
    </strong>
    /
    <strong>
       
      <a
        class="c2"
        href="https://github.com/facebook/react"
      >
        react
      </a>
       
    </strong>
    <label
      class="c3"
    >
      Public
    </label>
  </div>
</div>
`;
